### PR TITLE
Implement quick brick version

### PR DESCRIPTION
**Pull request by VladSumtsov**
## Description
- Implement QuickBrick redirect plugin
- Change android only repo to the multi repo.

## Checklist

* [ ] PR is scoped to one task
* [ ] Tests are included
* [ ] Documentation is included

* [ ] This PR is not making any UI change
* [ ] This PR is making a UI change
  * [ ] Screenshot(s) included

## QA :

* required test cases:
* specific apps / plugins to test :

### CODE REVIEW

File: android/local.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## This file must *NOT* be checked into Version Control Systems,
```
Comment (guysarfatty) [inline android/local.properties]: This file needs to be deleted

File: QuickBrick/manifest/tvos.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (guysarfatty) [inline QuickBrick/manifest/tvos.json]: How is this related to tvOS? I don't think this should be checked in

File: android/app/src/androidTest/java/com/applicaster/android_url_redirection_plugin/.idea/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Comment (guysarfatty) [inline android/app/src/androidTest/java/com/applicaster/android_url_redirection_plugin/.idea/.gitignore]: Please remove

File: android/app/src/androidTest/java/com/applicaster/android_url_redirection_plugin/.idea/vcs.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (guysarfatty) [inline android/app/src/androidTest/java/com/applicaster/android_url_redirection_plugin/.idea/vcs.xml]: Please remove

File: android/app/src/androidTest/java/com/applicaster/android_url_redirection_plugin/.idea/modules.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (guysarfatty) [inline android/app/src/androidTest/java/com/applicaster/android_url_redirection_plugin/.idea/modules.xml]: please remove

File: QuickBrick/manifest/tvos.json (hunk 2 above)
Comment (VladSumtsov) [reply QuickBrick/manifest/tvos.json]: This is QuickBrick redirect URL plugin. So I published two manifests for tvOS and AndroidTV. This plugin uses for these two platforms here: https://zapp.applicaster.com/app_families/1558

Comment (guysarfatty) [reply QuickBrick/manifest/tvos.json]: Why would you need it on tvOS though? On iOS and tvOS we don't have issues with url redirection.. or were there? If it is also a problem in tvOS, we might need to change the repo, as it is currently referring to Android only

File: README.md [hunk 6]
```diff
@@ -1,5 +1,8 @@
 
-# android-url-redirection-plugin
+This is multiplatform repository for redirect url plugins.
```
Comment (guysarfatty) [inline README.md]: I am not sure that this is where I would put this description. I would put it after "This repository contains the code for the plugin that handles redirection of stream urls on Android". 
The following line: "The repository contains the code for Android mobile and AndroidTV via Quickbrick".

File: QuickBrick/manifest/tvos.json (hunk 2 above)
Comment (VladSumtsov) [reply QuickBrick/manifest/tvos.json]: I didn't know about this. I had the task to write redirect URL on QuickBrick. Should I clarify this requirement?

Comment (VladSumtsov) [reply QuickBrick/manifest/tvos.json]: Yeap I will remove this. You are right.

File: QuickBrick/plugin/src/index.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      .then(function(response) {
+        const responseUrl = R.path(["request", "responseURL"], response);
+        const redirectedPayload = R.set(videoSourcePath, responseUrl, payload);
+        callback({ success: true, payload: redirectedPayload })
+      })
+      .catch(function(error) {
+        callback({ success: false, error })
+      })
+      .finally(function() {
+      });
```
Comment (oduk-sc) [inline QuickBrick/plugin/src/index.js]: Seems that `finally` block is unnecessary here

File: QuickBrick/plugin/src/index.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  callback: ({ success: boolean, error: ?{}, payload: ?{} }) => void
+};
+
+type State = {}
+
+const videoSourcePath = R.lensPath(["content", "src"]);
+
+export class RedirectUrlComponent extends React.Component<Props, State> {
+
+  state = {};
```
[the remark this answers is not in the document]
Comment (VladSumtsov) [reply QuickBrick/plugin/src/index.js]: What do you mean?

[the remark this answers is not in the document]
Comment (oduk-sc) [reply QuickBrick/plugin/src/index.js]: If you don't need the state of the component, I'd advise you to declare this component  as a function component:
```js
function RedirectUrlComponent(props) {
   useEffect(() => {
     // here you can perform requests like you did in componentDidMount lifecycle method
    }, [])
    return <View></View>
}
```

[the remark this answers is not in the document]
Comment (VladSumtsov) [reply QuickBrick/plugin/src/index.js]: Thanks! :)

[the remark this answers is not in the document]
Comment (VladSumtsov) [reply QuickBrick/plugin/src/index.js]: In the root index.js file we need to add this component as a component, it doesn't accept functions. So, we can't use the proposed way here.

File: QuickBrick/plugin/index.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import {RedirectUrlComponent} from "./src"
+
+const RedirectUrlPlugin = {
+  presentFullScreen: true,
+  hasPlayerHook: true,
+  Component: RedirectUrlComponent
+};
```
Comment (oduk-sc) [inline QuickBrick/plugin/index.js]: Seems that you removed isFlowBlocker key, please re-add it here (it was not needed in manifest because you had it here in your root index.js)
```js
const RedirectUrlPlugin = {
  presentFullScreen: true,
  isFlowBlocker: () => true,
  hasPlayerHook: true,
  Component: RedirectUrlComponent
};
```

File: QuickBrick/plugin/src/index.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (oduk-sc) [reply QuickBrick/plugin/src/index.js]: your root index.js does not depend on the type of component you are using (functional or class).  You can just leave you root index.js as
```js
const RedirectUrlPlugin = {
  presentFullScreen: true,
  isFlowBlocker: () => true,
  hasPlayerHook: true,
  Component: RedirectUrlComponent
};
```
and refactor your RedirectUrlComponent class component into a functional component, as I mentioned above

**Comment by oduk-sc** [comment]
LGTM
